### ISSUE TITLE
[Windows Client] Entry Point Error

### ISSUE THREAD

**Issue by zQueal** [keybase/client#2498]
Recently I had to reformat my PC due to various errors after not getting KBFS to work again because of "unsigned drivers". (#2418) After reformatting and installing the version provided by @zanderz it was working fine again and all was well in the Kingdom. Yesterday I got a notification to update my client. Which I did. And the error resurfaced. Basically, this time I'm not going to mess with it for obvious reasons.

Instead, I opted to uninstall the new version and re-install the older version which worked fine. Upon doing so I got this error while installing: 
![cfpztbr](https://cloud.githubusercontent.com/assets/239193/14164342/bfd58a76-f6cc-11e5-929e-ee3b0859caf0.png)

The install went fine, however, because of this error I'm unable to use the KBFS, or KBFS GUI. CLI works just fine, it seems.

-------------------------------------------

After a bunch of mucking about, it would seem that during the `keybase update` process, the `Dokan` services was never properly removed so it could be reinstalled. This caused the new client (or windows?) to simply install the new service as `Dokan1` which isn't correct for the signature (`Dokan` is valid, not `Dokan1`) making it invalid and Windows to not recognize it as a validly signed service prompting the errors I was having and stopping the KBFS from ever running. After manually stopping, disabling, and then removing the `Dokan1` service along with `DokanMounter`, restarting, reinstalling Keybase, and then restarting again. I'm up and running again.

The issue is, however, when I go to update next, will I have the same issues with the Dokan service possibly being in use at the time of uninstall leaving it in the Windows service registry prompting the same issues as before.

**Comment by zanderz** [comment]
Zach,
There are several moving pieces here. First thing is, yes, the change from "dokan" to "dokan1" is major and complicated - so much so that the latest installers first attempt to perform an uninstall behind the scenes, then prompt to reboot before continuing. If you are running an older Keybase KBFS alpha with dokan.sys and get an update to one with dokan1.sys, you should be prompted to reboot during its installation, otherwise it is a bug. One way to see what is going on is to launch the updated installer yourself with logging activated, as in:
`keybase_setup_gui_XXXXX.386.exe /LOG=%userprofile%\Documents\update.log`

Next, the signing of dokan1.sys: It is signed. If you navigate to it in Explorer and examine its properties, you should be able to see it as in the attached. The name of the file (dokan or dokan1) is not relevant. 
![image](https://cloud.githubusercontent.com/assets/862397/14182688/4a96ec6e-f721-11e5-8a07-6c7c5a18a495.png)

What may be relevant is that Microsoft deprecated SHA1 codesigning on Windows 7 at the start of this year. It is a complicated and confusing subject, but it appears that an unpatched Windows 7 machine that is unable to download intermediate certificates from the internet may not be able to verify code signed with newer certs. There is lots of interesting information on this subject at http://www.davidegrayson.com/signing and elsewhere. We are working on reproducing this environment and may be able to eventually offer a workaround.

Are you certain your Windows 7 is updated?

**Comment by zQueal** [comment]
Damn! I was positive I did some awesome snooping there for a minute. :(

I read the entirety of [davidegrayson.com](http://www.davidegrayson.com/signing) and ended up tracking down the update needed which is [KB3033929](https://www.microsoft.com/en-us/download/details.aspx?id=46148). I installed the update. Restarted. Ran `keybase update check` and installed the new update. I was prompted to restart and I did. Upon restart, everything is working the same as it once was on the newest version;

    λ keybase version
    Client:  1.0.14-20160329160927+76dfe98
    Service: 1.0.14-20160329160927+76dfe98

Sorry for the noise, and thanks for the help!
